blocks/carousel/carousel.js

```diff
@@ -6,6 +6,63 @@ Modifications (preferably avoid) may have been made to fit the specific needs of
 */
 import { fetchPlaceholders } from '../../scripts/aem.js';
 
+/** Extends the block with additional functionality.
+ * @param block - the carousel block
+ * @returns {{autoScrollDelayInMilliseconds: (number|number), isAutoScroll: boolean,
+ * isMultiSlide: boolean, autoScrollCount: number, visibleSlides: (number|number)}}
+ */
+function extendedFunctionality(block) {
```

Comment:
@shrotia should we rename this function to use case something like `addingAutoScroll` ?

Reply:
Not just the AutoScroll, this method takes care of instantiating all the fields that adds custom behaviour on top of Carousel. For example `isMultiSlide` determines if the Carousel will have multiple cards or a single card
I have just pulled this method and named `extendedFunctionality` to avoid mixing extensions with OOTB carousel features.